src/components/donation-flow/DonationFlowForm.tsx

```diff
@@ -4,103 +4,97 @@ import { useElements, useStripe } from '@stripe/react-stripe-js'
-  amount?: string
-  amountWithFees?: number
-  otherAmount?: number
+const generalValidation = {
+  payment: yup
+    .string()
+    .oneOf(Object.values(DonationFormDataPaymentOption))
+    .required() as yup.SchemaOf<DonationFormDataPaymentOption>,
+  authentication: yup
+    .string()
+    .oneOf(Object.values(DonationFormDataAuthState))
+    .required() as yup.SchemaOf<DonationFormDataAuthState>,
+  isAnonymous: yup.boolean().required(),
+  email: yup
+    .string()
+    .email('one-time-donation:errors-fields.email')
+    .required()
+    .when('authentication', {
+      is: 'NOREGISTER',
+      then: yup.string().email('one-time-donation:errors-fields.email').required(),
+    }),
+  privacy: yup.bool().required().isTrue('one-time-donation:errors-fields.privacy'),
 }
 
 const initialValues: DonationFormDataV2 = {
-  amount: '',
-  email: '',
-  payment: null,
-  amountWithFees: 0,
-  cardIncludeFees: false,
-  cardRegion: CardRegion.EU,
-  otherAmount: 0,
-  authentication: null,
-  isAnonymous: false,
+  ...initialGeneralFormValues,
+  ...initialAmountFormValues,
+  ...initialLoginFormValues,
+  ...initialRegisterFormValues,
 }
 
 export const validationSchema: yup.SchemaOf<DonationFormDataV2> = yup
   .object()
   .defined()
   .shape({
-    payment: yup
-      .string()
-      .oneOf(Object.values(DonationFormDataPaymentOption))
-      .required() as yup.SchemaOf<DonationFormDataPaymentOption>,
-    amount: yup.string().when('payment', {
-      is: 'card',
-      then: yup.string().required(),
-    }),
-    amountWithFees: yup.number().when('payment', {
-      is: 'card',
-      then: () =>
-        yup.number().min(1, 'one-time-donation:errors-fields.amount-with-fees').required(),
-    }),
-    otherAmount: yup.number().when('amount', {
-      is: 'other',
-      then: yup.number().min(1, 'one-time-donation:errors-fields.other-amount').required(),
-    }),
-    cardIncludeFees: yup.boolean().required(),
-    cardRegion: yup
-      .string()
-      .oneOf(Object.values(CardRegion))
-      .when('payment', {
-        is: 'card',
-        then: yup.string().oneOf(Object.values(CardRegion)).required(),
-      }) as yup.SchemaOf<CardRegion>,
-    authentication: yup
-      .string()
-      .oneOf(Object.values(DonationFormDataAuthState))
-      .required() as yup.SchemaOf<DonationFormDataAuthState>,
-    isAnonymous: yup.boolean().required(),
-    email: yup
-      .string()
-      .required()
-      .when('authentication', {
-        is: 'NOREGISTER',
-        then: yup.string().email('one-time-donation:errors-fields.email').required(),
-      }),
+    ...generalValidation,
+    ...amountValidation,
+    ...loginValidation,
+    ...registerFormValidation,
   })
 
 export function DonationFlowForm() {
   const { data: session } = useSession()
-  const { campaign, stripePaymentIntent } = useContext(DonationFlowContext)
+  const { campaign, stripePaymentIntent, paymentError, setPaymentError } =
+    useContext(DonationFlowContext)
```

Comment:
We can add a helper hook to get the context in a seamless way with:

```tsx
export function useDonationFlow() {
  return useContext(DonationFlowContext)
}
```

```suggestion
  const { campaign, stripePaymentIntent, paymentError, setPaymentError } = useDonationFlow()
```

Reply:
Done in [d45154f](https://github.com/podkrepi-bg/frontend/pull/1314/commits/d45154fb0795cde4f48321470dda0b24c2940d51)